### ISSUE TITLE
VOCMApMetric not counting False Positives

### ISSUE THREAD

**Issue by oneluckypic** [dmlc/gluon-cv#1161]
Test:
```python
from gluoncv.utils.metrics.voc_detection import VOCMApMetric
import mxnet as mx
import numpy as np
def test_voc_map_metric():
    def w(a):
        return [mx.nd.array([a])]
    p1 = [0, 0, 10, 10]
    p2 = [10, 10, 20, 20]
    pred_bboxes = w([p1, p2])
    pred_labels = w([0, 0])
    pred_scores = w([.5, .5])
    gt_bboxes = w([p1])
    gt_labels = w([0])
    gt_difficults = w([0])
    metric = VOCMApMetric(class_names=['dog'])
    metric.update(pred_bboxes, pred_labels, pred_scores,
                      gt_bboxes, gt_labels, gt_difficults)
    print(metric.get())
```
Actual Result:
(['dog', 'mAP'], [1.0, 1.0])

Expected Results:
I would expect 0.5 for both 'dog' and 'mAP'.  Precision is TP / (TP + FP).  There is one TP and one FP, i.e. (1 / (1 + 1).

**Comment by oneluckypic** [comment]
Okay, I've convinced myself this is working properly.

The mAP is the area under the precision/recall curve.  In the example above the predictions are sorted by confidence value.  Since they are tied at .5 it happens that p1 is processed first.  p1 is the only truth labels which immediately puts the recall at 1 and precision at 1.  Next p2 is looked at, the recall still remains 1 but the precision goes down to 0.5.  It doesn't matter though because the recall was already pushed to 1 and the P/R curve just dives vertically down to 0.5 which doesn't change the area under the curve.

Note that if p2 is processed before p1 the mAP would be 0.5, but this is because they are a tie.  The likelihood of ties happening in the logits is I'm sure very low and random, so it doesn't matter much.

So long as all of your True Positives have higher confidence than your False Positive and you find all the Ground Truth labels in your images (recall 1.0), your mAP will be 1.0.  This was somewhat unexpected to me, but makes sense.
